redwood-lazylayout-compose/src/commonMain/kotlin/app/cash/redwood/lazylayout/compose/LazyList.kt

```diff
@@ -45,24 +45,24 @@ internal fun LazyList(
   content: LazyListScope.() -> Unit,
 ) {
   val itemProvider = rememberLazyListItemProvider(content)
-  var lastVisibleItemIndex by remember { mutableStateOf(0) }
+  var lastPagedItemIndex by remember { mutableStateOf(0) }
   val itemsBefore = remember(state.firstVisibleItemIndex) { (state.firstVisibleItemIndex - OffscreenItemsBufferCount / 2).coerceAtLeast(0) }
-  val itemsAfter = remember(lastVisibleItemIndex, itemProvider.itemCount) { (itemProvider.itemCount - (lastVisibleItemIndex + OffscreenItemsBufferCount / 2).coerceAtMost(itemProvider.itemCount)).coerceAtLeast(0) }
+  val itemsAfter = remember(lastPagedItemIndex, itemProvider.itemCount) { (itemProvider.itemCount - (lastPagedItemIndex + OffscreenItemsBufferCount / 2).coerceAtMost(itemProvider.itemCount)).coerceAtLeast(0) }
   val scrollItemIndex = remember(state.scrollToItemTriggeredId) { ScrollItemIndex(state.scrollToItemTriggeredId, state.firstVisibleItemIndex) }
   var placeholderPoolSize by remember { mutableStateOf(20) }
   LazyList(
     isVertical,
     itemsBefore = itemsBefore,
     itemsAfter = itemsAfter,
-    onViewportChanged = { localFirstVisibleItemIndex, localLastVisibleItemIndex ->
-      val visibleItemCount = localLastVisibleItemIndex - localFirstVisibleItemIndex
+    onViewportChanged = { localFirstPagedItemIndex, localLastPagedItemIndex ->
+      val visibleItemCount = localLastPagedItemIndex - localFirstPagedItemIndex
       val proposedPlaceholderPoolSize = visibleItemCount + visibleItemCount / 2
       // We only ever want to increase the pool size.
       if (placeholderPoolSize < proposedPlaceholderPoolSize) {
         placeholderPoolSize = proposedPlaceholderPoolSize
       }
-      state.firstVisibleItemIndex = localFirstVisibleItemIndex
-      lastVisibleItemIndex = localLastVisibleItemIndex
+      state.firstVisibleItemIndex = localFirstPagedItemIndex
```

Comment:
@swankjesse @JakeWharton I didn't notice it before, but such a paginated viewport introduces a bug here. This PR means we can no longer reliably expose to the caller of `LazyList` what the first visible item index is.

Reply:
Can we add a third parameter to the lambda for the actual first index?

Reply:
If we did that and didn't update the frequency of the lambda invocation on every scroll, then the first visible item index would still be incorrect due to the new deduplication logic